### PR TITLE
Raster bug fix

### PR DESCRIPTION
**Pull request by gantian127**
fix the bugs with raster 
#1162 #1335 #1367

### CODE REVIEW

**Comment by hydrobuild** [comment]
Build finished.

File: hs_geo_raster_resource/raster_meta_extract.py [hunk 1]
```diff
@@ -251,7 +252,18 @@ def get_band_info(raster_file_name):
...
+            minimum, maximum, _, _ = band.ComputeStatistics(False)
+            no_data = band.GetNoDataValue()
+            new_no_data = None
+
+            if no_data and numpy.allclose(minimum, no_data):
+                new_no_data = minimum
+            elif no_data and numpy.allclose(maximum, no_data):
+                new_no_data = maximum
+
+            if new_no_data:
```
Comment (pkdash) [inline hs_geo_raster_resource/raster_meta_extract.py]: Should use ```if new_no_data is not None:```

File: hs_geo_raster_resource/receivers.py [hunk 2]
```diff
@@ -83,11 +83,27 @@ def raster_file_validation(files, ref_tmp_file_names=[]):
...
-                             ' with the references in the .vrt file.')
+
+        if len(files_names) > len(raster_file_names):
+            error_info.append('Please remove the extra raster files which are not specified in the .vrt file.')
+        else:
+            for vrt_ref_raster_name in raster_file_names:
+                if vrt_ref_raster_name in files_names \
+                        or (os.path.split(vrt_ref_raster_name)[0] == '.' and
+                            os.path.split(vrt_ref_raster_name)[1] in files_names):
+                    pass
```
Comment (pkdash) [inline hs_geo_raster_resource/receivers.py]: Use ```continue``` in place of ```pass```

**Comment by pkdash** [comment]
@gantian127 Needs resolving the conflicts.

**Comment by mjstealey** [comment]
This PR is being run manually here: [http://ci.hydroshare.org:8080/job/github-pr-builder-hydroshare/302/](http://ci.hydroshare.org:8080/job/github-pr-builder-hydroshare/302/)

Due to the install of GDAL, the prior build had hit the timeout flag prior to completing. The timeout flag has been increased to accommodate this build.

**Comment by pkdash** [comment]
+1

**Pull request merged**